x-pack/platform/plugins/shared/stack_connectors/server/connector_types/microsoft_defender_endpoint/microsoft_defender_endpoint.ts

```diff
@@ -110,19 +109,45 @@ export class MicrosoftDefenderEndpointConnector extends SubActionConnector<
   ): Promise<R> {
     this.logger.debug(() => `Request:\n${JSON.stringify(req, null, 2)}`);
 
-    const bearerAccessToken = await this.oAuthToken.get(connectorUsageCollector);
-    const response = await this.request<R>(
-      {
-        ...req,
-        // We don't validate responses from Microsoft API's because we do not want failures for cases
-        // where the external system might add/remove/change values in the response that we have no
-        // control over.
-        responseSchema:
-          MicrosoftDefenderEndpointDoNotValidateResponseSchema as unknown as SubActionRequestParams<R>['responseSchema'],
-        headers: { Authorization: `Bearer ${bearerAccessToken}` },
+    const requestOptions: SubActionRequestParams<R> = {
+      ...req,
+      // We don't validate responses from Microsoft API's because we do not want failures for cases
+      // where the external system might add/remove/change values in the response that we have no
+      // control over.
+      responseSchema:
+        MicrosoftDefenderEndpointDoNotValidateResponseSchema as unknown as SubActionRequestParams<R>['responseSchema'],
+      headers: {
+        Authorization: `Bearer ${await this.oAuthToken.get(connectorUsageCollector)}`,
       },
-      connectorUsageCollector
-    );
+    };
+    let response: AxiosResponse<R>;
+    let was401RetryDone = false;
+
+    try {
+      response = await this.request<R>(requestOptions, connectorUsageCollector);
+    } catch (err) {
+      if (was401RetryDone) {
+        throw err;
+      }
+
+      this.logger.debug("API call failed! Determining if it's one we can retry");
+
+      // If error was a 401, then for some reason the token used was not valid (ex. perhaps the connector's credentials
+      // were updated). IN this case, we will try again by ensuring a new token is re-generated
+      if (err.message.includes('Status code: 401')) {
```

Comment:
How sure are we that 401 is returned only in this specific case? 
Or on second though - it doesn't hurt us to retry anyway :) 
Thanks!

Reply:
We return this error - its built by the `.getResponseErrorMessage()` method of this class. Unfortunately the connector framework does not return the original error (`Axios`) from the `.request()`, thus why we are relying on the `.message` property